### PR TITLE
sensors: max17262: Error-check init I2C operations

### PR DESCRIPTION
**Pull request by tristan-google**
The init function of the MAX17262 sensor doesn't check the return value
of its I2C read and write functions. In case a read operation fails, the
output variable is not updated but the driver proceeds anyways. This can
cause unintended operation due to the potentially un-initialized memory,
such as getting stuck in the polling loop on line `max17262.c:245`. Update
the init function to abort and pass along the error code when I2C
transactions fail.

Also run clang-format on the file (see separate commit)

Signed-off-by: Tristan Honscheid <honscheid@google.com>

### CODE REVIEW

**Review by yperess — DISMISSED** [review]

**Review by avisconti — DISMISSED** [review]
LGTM

**Review by fabiobaltieri — DISMISSED** [review]

File: drivers/sensor/max17262/max17262.c [hunk 1]
```diff
@@ -178,26 +175,19 @@ static int max17262_channel_get(const struct device *dev,
...
-		{ TTF, &data->time_to_full },
-		{ CYCLES, &data->cycle_count },
-		{ DESIGN_CAP, &data->design_cap },
-		{ COULOMB_COUNTER, &data->coulomb_counter },
+		{VCELL, &data->voltage},          {AVG_CURRENT, &data->avg_current},
+		{ICHG_TERM, &data->ichg_term},    {REP_SOC, &data->state_of_charge},
+		{INT_TEMP, &data->internal_temp}, {REP_CAP, &data->remaining_cap},
+		{FULL_CAP_REP, &data->full_cap},  {TTE, &data->time_to_empty},
+		{TTF, &data->time_to_full},       {CYCLES, &data->cycle_count},
+		{DESIGN_CAP, &data->design_cap},  {COULOMB_COUNTER, &data->coulomb_counter},
```
Comment (fabiobaltieri) [inline drivers/sensor/max17262/max17262.c:190]: This makes no sense at all, then if any of the left side changes it's likely to cause the whole block to be realigned, causing even more unnecessary diffs, potentially making other changes harder to spot in code reviews etc. I'd suggest adding a `clang-format off` around this if you really want to reformat the file but really I don't think we should attach file-wide reformat next to bugfixes. I could understand if the file had a major refactoring but this is just adding noise.

Comment (tristan-google) [reply drivers/sensor/max17262/max17262.c:190]: I have the reformatting as a separate commit and could drop that if you want.

Comment (fabiobaltieri) [reply drivers/sensor/max17262/max17262.c:190]: Up to you whether to keep it, I would not bother unless I was doing heavy rework on the driver, but if you keep it I'd ask you to block this one off.

**Review by fabiobaltieri — DISMISSED** [review]

File: drivers/sensor/max17262/max17262.c (hunk 1 above)
Comment (tristan-google) [reply drivers/sensor/max17262/max17262.c]: Excluded this portion from formatting.

**Review by fabiobaltieri — APPROVED** [review]

**Review by yperess — APPROVED** [review]

**Pull request merged**